lang/driver/src/database.rs

```diff
@@ -331,7 +331,8 @@ impl Database {
 
     pub fn print_to_string(&mut self, uri: &Url) -> Result<String, Error> {
         let module = self.load_ast(uri)?;
-        let module = (*module).clone().rename();
+        let mut module = (*module).clone();
+        module.rename();
```

Comment:
Maybe we should instead guarantee that the cached AST is always renamed such that it can be printed immediately. I.e. maybe we should move the renaming to `load_ast`. This would also get rid of this clone here.

Reply:
I think this might interact with the hover-information which displays contexts. I think the hover information is also computed from the variable names that are annotated in the AST, and if we rename before we compute the hover information then the type-on-hover might show variable names that do not occur in the source text.

Reply:
Good point, although that should be resolved by building a smarter renaming that only renames when absolutely necessary (e.g. allowing duplicate names when a shadowed name is not referenced in the respective scope and always preferring the user-provided names).